docs/MAZERUNNER.md

```diff
@@ -0,0 +1,48 @@
+# Mazerunner tests
+
+E2E tests are implemented with our notifier testing tool [Maze runner](https://github.com/bugsnag/maze-runner), which is a black-box test framework written in Ruby.
+
+## Running remote end-to-end tests
+
+End to end tests are written in cucumber-style `.feature` files, and need Ruby-backed "steps" in order to know what to run. The tests are located in the top level [`features`](/features/) directory.
+
+Maze Runner's CLI and the test fixtures are containerized so you'll need Docker (and Docker Compose) to run them.
+
+__Note: only Bugsnag employees can run the end-to-end tests.__ We have dedicated test infrastructure and private BrowserStack credentials which can't be shared outside of the organisation.
+
+### Authenticating with the private container registry
+
+You'll need to set the credentials for the aws profile in order to access the private docker registry:
+
+```
+aws configure --profile=opensource
+```
+
+Subsequently you'll need to run the following commmand to authenticate with the registry:
+
+```
+$(aws ecr get-login --profile=opensource --no-include-email)
+```
+
+__Your session will periodically expire__, so you'll need to run this command to re-authenticate when that happens.
+
+Remote tests can be run against real devices provided by BrowserStack. In order to run these tests, you need to set the following environment variables:
+
+- A BrowserStack App Automate Username: `MAZE_DEVICE_FARM_USERNAME`
+- A BrowserStack App Automate Access Key: `MAZE_DEVICE_FARM_ACCESS_KEY`
+- A path to a [BrowserStack local testing binary](https://www.browserstack.com/local-testing/app-automate): `MAZE_BS_LOCAL`
+
+### Running an end-to-end test
+
+1. Build the test fixtures `make test-fixtures` (separate fixtures are built with and without the NDK/ANR plugins)
+1. Check the contents of `Gemfile` to select the version of `maze-runner` to use
+1. To run a single feature:
+    ```shell script
+    make test-fixtures && \
```

Comment:
This duplicates the first step.

Reply:
I always forget to run `make test-fixtures` and get confused with the results. I've therefore removed the first step and kept the command as-is

Reply:
I guess that's fine, it's just potentially a waste of time for people who run it without thinking about what it's actually doing.  If you're just running different tests without changing the notifier or scenario classes then you don't need to rebuild it - but I can live with it as it is now!